addons/web/static/src/views/fields/properties/property_tags.js

```diff
@@ -0,0 +1,237 @@
+                        || tag[1].toLocaleLowerCase().indexOf(request.toLocaleLowerCase()) >= 0
+                    )
+                );
+                if (!tagsFiltered || !tagsFiltered.length) {
+                    // no result, ask the user if he want to create a new tag
+                    if (!request || !request.length) {
+                        return [{
+                            value: null,
+                            label: _lt('Start typing...'),
+                            classList: 'fst-italic',
+                        }];
+                    } else if (!this.props.canChangeTags) {
+                        return [{
+                            value: null,
+                            label: _lt('No result'),
+                            classList: 'fst-italic',
+                        }];
+                    }
+
+                    return [{
+                        value: {toCreate: true, value: request},
+                        label: sprintf(_lt('Create "%s"'), request),
+                        classList: 'o_field_property_dropdown_add',
+                    }];
+                }
+                return tagsFiltered.map((tag) => {
+                    return {
+                        value: tag[0],
+                        label: tag[1],
+                    };
+                });
+            },
+        }];
+    }
+
+    /**
+     * Add one value in the current tag list values.
+     */
+    onOptionSelected(tagValue) {
+        if (!tagValue) {
+            // clicked on "Start typing..."
+            return;
+        }
+
+        if (tagValue.toCreate) {
+            this.onNewTag(tagValue.value);
+        } else {
+            const currentValue = this.currentValue;
+            const newValue = [...currentValue, tagValue];
+            this.props.onValueChange(newValue);
+        }
+    }
+
+    /**
+     * Ask to create a new tag that will be added in
+     * the definition and automatically selected.
+     */
+    onNewTag(newLabel) {
+        if (!newLabel || !newLabel.length) {
+            return;
+        }
+
+        const newValue = this.labelToValue(newLabel);
+
+        const existingTags = this.props.tags.find(tag => tag[0] === newValue);
+        if (existingTags) {
+            this.notification.add(
+                _lt('This tag is already available'),
+                { type: 'warning' },
+            );
+            return;
+        }
+
+        const tagColor = parseInt(Math.random() * ColorList.COLORS.length);
+        const newTag = [newValue, newLabel, tagColor];
+        const newTags = [...this.currentTags, newTag];
+        // automatically select the newly created tag
+        const newValues = [...this.props.value, newTag[0]];
+        this.props.onTagsChange(newTags, newValues);
+    }
+
+    /**
+     * Click on the delete button on the tag pill.
+     * The behavior is defined by the prop "deleteAction".
+     */
+    onTagDelete(deleteTag) {
```

Comment:
I'm guessing this is because you use the same component for both the "configuration" and the "selection" of tags?

Should be explained a bit more.

I'm hesitating about having a base Component that defines method and then 2 different implementations, but that might be overkill as it seems there are very little differences.